arbor/hardware/node_info.cpp

```diff
@@ -1,19 +1,51 @@
-#include <algorithm>
+#include <thread>
+
+#ifdef ARB_HAVE_GPU
+#include <cuda_runtime.h>
+#endif
+
+// TODO: C++17 use __has_include(<unistd.h>)
+#if defined(__unix__) || defined(__APPLE__) && defined(__MACH__)
+#include <unistd.h>
+#endif
 
 #include "affinity.hpp"
-#include "gpu.hpp"
 #include "node_info.hpp"
 
 namespace arb {
 namespace hw {
 
-// Return a node_info that describes the hardware resources available on this node.
-// If unable to determine the number of available cores, assumes that there is one
-// core available.
-node_info get_node_info() {
-    auto res = num_cores();
-    unsigned ncpu = res? *res: 1u;
-    return {ncpu, num_gpus()};
+unsigned node_gpus() {
+#ifdef ARB_HAVE_GPU
+    int n;
+    if (!cudaGetDeviceCount(&n)) {
+        return static_cast<unsigned>(n);
```

Comment:
You use `static_cast<unsigned>` here, and `(unsigned)` below.
I am comfortable with using the second option.

Reply:
Cool beans, will change.